test/controllers/configs_controller_test.rb

```diff
@@ -1,9 +1,42 @@
 require 'test_helper'
 
 class ConfigsControllerTest < ActionDispatch::IntegrationTest
-  test "should get index" do
-    get configs_index_url
-    assert_response :success
+  describe 'non-admin redirects' do
+    [:data_volunteer, :cm].each do |role|
+      it "should deny access as a #{role}" do
+        delete destroy_user_session_path
+        user = create :user, role: role
+        config = create :config
+        sign_in user
+
+        get configs_path
+        assert_redirected_to authenticated_root_path
+
+        patch config_path(config, params: { options: ['no'] })
```

Comment:
Why does this test setup on the negative test look different that the setup on the positive test down on line 35 onwards? i.e. `config` vs `@config`. `params: { options: ['no'] }` vs `params: { config: { options: ['no']}}`